server.go

```diff
@@ -62,28 +62,100 @@ func (r *Runner) EnsureYumUpdated() error {
 }
 
 func (r *Runner) InstallGem(p string) error {
-	err := r.EnsureRubyInstalled()
+	if !r.rvmInstalled {
+		err := r.EnsureRvmInstalled()
+		if err != nil {
+			return err
+		}
+		r.rvmInstalled = true
+	}
+	err := r.Command("gem", "install", p)
 	if err != nil {
 		return err
 	}
-	err = r.Command("gem", "install", p)
+	return nil
+}
+
+func (r *Runner) curlRvm(p string) error {
+	err := r.Command("curl", p)
 	if err != nil {
 		return err
 	}
 	return nil
 }
 
-func (r *Runner) EnsureRubyInstalled() error {
-	if !r.rubyInstalled {
-		err := r.InstallPackage("epel-release")
-		if err != nil {
-			return err
-		}
-		err = r.InstallPackage("ruby")
-		if err != nil {
-			return err
-		}
-		r.rubyInstalled = true
+func (r *Runner) sourceRvm(p string) error {
+	err := r.Command("source", p)
+	if err != nil {
+		return err
+	}
+	return nil
+}
+
+func (r *Runner) installRvm(p string) error {
+	err := r.Command("rvm", p)
+	if err != nil {
+		return err
+	}
+	return nil
+}
+
+func (r *Runner) EnsureRvmInstalled() error {
+
+	var rvmPackages string = "epel-release gcc-c++ patch readline readline-devel zlib zlib-devel libffi-devel openssl-devel make bzip2 autoconf automake libtool bison sqlite-devel"
```

Comment:
We might make `rvmDependencies` (a better name, no?) a package-level constant. It's the kind of thing we're likely to need to change in the future, so putting it near the top of the package will help with this.

Reply:
good point